### PR TITLE
Add adjacent samples API endpoint

### PR DESCRIPTION
**Pull request by horatiualmasan**
## What has changed and why?

New endpoint for adjacent samples

## How has it been tested?

new tests

## Did you update [CHANGELOG.md](../CHANGELOG.md)?

- [ ] Yes
- [x] Not needed (internal change)

### CODE REVIEW

File: lightly_studio/tests/api/routes/api/test_adjacents.py [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+
+def test_get_adjacent_samples__missing_body_returns_unprocessable(
+    test_client: TestClient,
+) -> None:
+    response = test_client.post(
+        f"/api/samples/{uuid4()}/adjacents",
+        json={},
+    )
+
+    assert response.status_code == HTTP_STATUS_UNPROCESSABLE_ENTITY
```
Comment (michal-lightly) [inline lightly_studio/tests/api/routes/api/test_adjacents.py:169]: Please add a test for the case when `get_adjacent_samples` returns None (you need to ask for a sample id (can be made up I think) that is not in the collection.
